### ISSUE TITLE
Incomplete documentation of `while let` syntax regarding lazy boolean operator expressions.

### ISSUE THREAD

**Issue by steffahn** [rust-lang/reference#771]
Comparing the reference’s [section on `if let`](https://doc.rust-lang.org/nightly/reference/expressions/if-expr.html#if-let-expressions) vs. [the one on `while let`](https://doc.rust-lang.org/nightly/reference/expressions/loop-expr.html#predicate-pattern-loops), the latter contains no note about top-level `||` and `&&` being disallowed, even though this is the case by [the corresponding RFC](https://github.com/rust-lang/rfcs/blob/master/text/2497-if-let-chains.md#rollout-plan-and-transitioning-to-rust-2018) as well as in practice.

**Comment by Centril** [comment]
The corresponding RFC has not been stabilized, the syntax `if p && let p = q { ... }` is unstable. The reference only documents stable Rust.

**Comment by steffahn** [comment]
@Centril No. I’m talking about the part of the RFC that makes Rust2018 disallow certain expressions. This rule, that `||` and `&&` cant appear inside `if let` or `while let` is part of stable Rust. It is even in the reference already, however only for `if let`, not for `while let`.

**Comment by Centril** [comment]
Oh you mean the `Expression except struct or lazy boolean operator expression` bit in if-let vs. while-let; yeah that needs to be updated. Want to send in a PR?

**Comment by steffahn** [comment]
Sure, I’ll do that. Playing around with this, there seems to be more to it though.. is seems as if the `=` changes precedence inside an `if let`. According to [the reference](https://doc.rust-lang.org/nightly/reference/expressions.html#expression-precedence) it is supposed to be binding weaker than `&&` and `||` and there’s also `..` and `..=` in between. Apparently the implementation of this part of the RFC changed things in a way that also disallows `..`: [see this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=78e90cc80d80640e6a860f03f629585c).
